File: php/features.php
<?php
declare(strict_types=1);

// RemoteOkJobs SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RemoteOkJobsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RemoteOkJobsBaseFeature();
            case "test":
                return new RemoteOkJobsTestFeature();
            default:
                return new RemoteOkJobsBaseFeature();
        }
    }
}

File: py/features.py
# RemoteOkJobs SDK feature factory

from feature.base_feature import RemoteOkJobsBaseFeature
from feature.test_feature import RemoteOkJobsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RemoteOkJobsBaseFeature(),
        "test": lambda: RemoteOkJobsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
